### ISSUE TITLE
Allow for only single call to `DisposeScope.dispose`

### ISSUE THREAD

**Issue by bartekpacia** [RobertOdrowaz/dispose-scope#21]
IMHO, DisposeScope should be single use – we should be able to `dispose()` it only once. Trying to `dispose`() more than once should trigger some assertion or throw an exception. 

[Similar safety check is already in place here](https://github.com/RobertOdrowaz/dispose-scope/blob/master/packages/dispose_scope/lib/src/disposed_extensions/dispose_scope_disposed.dart#L6-L13).

cc: @RobertOdrowaz

**Comment by bartekpacia** [comment]
Additionally, I'd introduce `DisposeScope.run`:


```dart
/// Calls [block] only if this dispose scope is not disposed.
  Future<void> run(Future<void> Function(DisposeScope scope) block) async {
    if (!_isDisposed) {
      await block(this);
    }
  }

  void disposed(DisposeScope disposeScope) {
    disposeScope.addDispose(dispose);
  }
```

**Comment by bartekpacia** [comment]
Fixed in #25.
